docs/user/security/securing-communications/index.asciidoc

```diff
@@ -4,127 +4,189 @@
+address. When choosing the SAN, you should pick whichever attribute you will be using to connect to {kib} in your browser, which is likely
+the FQDN.
 
-.. Set the certificate's `subjectAltName` to the hostname, fully-qualified domain name (FQDN), or IP address of the {kib} server.
 
-.. Set the certificate's Common Name (CN) to the {kib} server's hostname or FQDN. Using the server's IP address as the CN does not work.
-
-You may choose to generate a certificate and private key using {ref}/certutil.html[the {es} certutil tool]. If you already used certutil to
-generate a certificate authority (CA), you would generate a certificate/key for Kibana like so (using the `--dns` param to set the
-`subjectAltName`):
+You may choose to generate a certificate signing request (CSR) and private key using the {ref}/certutil.html[`elasticsearch-certutil`] tool.
+For example:
 
 [source,sh]
 --------------------------------------------------------------------------------
-bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12 --name kibana --dns localhost
+bin/elasticsearch-certutil csr -name kibana-server -dns some-website.com,www.some-website.com
 --------------------------------------------------------------------------------
 
-This will generate a certificate and private key in a PKCS #12 keystore named `kibana.p12`.
+This will produce a ZIP archive named `kibana-server.zip`. Extract that archive to obtain the PEM-formatted CSR (`kibana-server.csr`) and
+unencrypted private key (`kibana-server.key`). In this example, the CSR has a common name (CN) of `"kibana-server"`, a SAN of
+`"some-website.com"`, and another SAN of `"www.some-website.com"`.
 
+NOTE: You will need to use a certificate authority (CA) to sign your CSR to obtain your server certificate. This certificate's signature
+will be verified by web browsers that are configured to trust the CA.
 --
 
-. Enable TLS/SSL in `kibana.yml`:
+. Configure {kib} to access the server certificate and private key.
+
+.. If your server certificate and private key are in PEM format:
 +
 --
+Specify your server certificate and private key in `kibana.yml`:
+
 [source,yaml]
 --------------------------------------------------------------------------------
-server.ssl.enabled: true
+server.ssl.certificate: "/path/to/kibana-server.crt"
+server.ssl.key: "/path/to/kibana-server.key"
+--------------------------------------------------------------------------------
+
+If your private key is encrypted, add the decryption password to your <<secure-settings,{kib} keystore>>:
+
+[source,yaml]
+--------------------------------------------------------------------------------
+bin/kibana-keystore add server.ssl.keyPassphrase
 --------------------------------------------------------------------------------
 --
 
-. Specify your server certificate and private key in `kibana.yml`:
+.. Otherwise, if your server certificate and private key are contained in a PKCS#12 file:
```

Comment:
Likewise, I'd suggest changing this to a bulleted list and merged intro too:
```suggestion
** Otherwise, if your server certificate and private key are contained in a PKCS#12 file, specify the file path in `kibana.yml`
```